lib/oneview-sdk/resource/api300/c7000/scope.rb

```diff
@@ -0,0 +1,98 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+require_relative 'resource'
+
+module OneviewSDK
+  module API300
+    module C7000
+      # Scope resource implementation for API300 C7k
+      class Scope < Resource
+        BASE_URI = '/rest/scopes'.freeze
+
+        # Create a resource object, associate it with a client, and set its properties.
+        # @param [OneviewSDK::Client] client The client object for the OneView appliance
+        # @param [Hash] params The options for this resource (key-value pairs)
+        # @param [Integer] api_ver The api version to use when interracting with this resource.
+        #   Defaults to the client.api_version if it exists, or the OneviewSDK::Client::DEFAULT_API_VERSION.
+        def initialize(client, params = {}, api_ver = nil)
+          @data ||= {}
+          @data['type'] ||= 'Scope'
+          super
+        end
+
+        # Set data and save to OneView
+        # @param [Hash] attributes The attributes to add/change for this resource (key-value pairs)
+        # @raise [OneviewSDK::IncompleteResource] if the client or uri is not set
+        # @raise [StandardError] if the resource save fails
+        # @return [Resource] self
+        def update(attributes = {})
+          set_all(attributes)
+          ensure_client && ensure_uri
+          options = {
+            'If-Match' => @data['eTag'],
+            'body' => @data
+          }
+          response = @client.rest_put(@data['uri'], options, @api_version)
+          @client.response_handler(response)
+          self
+        end
+
+        # Delete resource from OneView
+        # @raise [OneviewSDK::IncompleteResource] if the client or uri is not set
+        # @return [true] if resource was deleted successfully
+        def delete
+          ensure_client && ensure_uri
+          response = @client.rest_delete(@data['uri'], { 'If-Match' => @data['eTag'] }, @api_version)
+          @client.response_handler(response)
+          true
+        end
+
+        # Adds resource assignments
+        # @param [Array] resources The array of resources (or resources separated by comma)
+        # @raise [OneviewSDK::IncompleteResource] if the client or uri is not set
+        def set_resources(*resources)
+          change_resources_assignments(add_resources: resources.flatten)
+        end
+
+        # Removes resource assignments
+        # @param [Array|Strings] resources The array of resources (or resources separated by comma)
```

Comment:
I'm working in that now. It will be `@param [Array]`. =)
But client can use any number of strings (without use of the array)